plugins/source/aws/client/testing.go

```diff
@@ -51,5 +51,46 @@ func AwsMockTestHelper(t *testing.T, table *schema.Table, builder func(*testing.
 		Version:      version,
 		Tables:       []string{table.Name},
 		Destinations: []string{"mock-destination"},
-	})
+	}, source.WithTestPluginAdditionalValidators(validateTagStructure))
+}
+func extractTables(tables schema.Tables) []*schema.Table {
+	result := make([]*schema.Table, 0)
+	for _, table := range tables {
+		result = append(result, table)
+		result = append(result, extractTables(table.Relations)...)
+	}
+	return result
+}
+
+func validateTagStructure(t *testing.T, plugin *source.Plugin, resources []*schema.Resource) {
+	for _, table := range extractTables(plugin.Tables()) {
+		t.Run(table.Name, func(t *testing.T) {
+			for i, column := range table.Columns {
+				if column.Name != "tags" {
+					continue
+				}
+				for _, resource := range resources {
+					if resource.Table.Name != table.Name {
+						continue
+					}
+
+					for iResource, value := range resource.GetValues() {
+						if iResource != i {
+							continue
+						}
+						if value.Get() != nil && value.Get() != schema.Undefined {
```

Comment:
Could this happen? Isn't it a problem if the value is nil or undefined?

Reply:
Right, but I'm saying it seems like this test should fail if the tags are nil or undefined, not succeed?

Reply:
Fair... I have removed that conditional